helix-term/src/ui/completion.rs

```diff
@@ -107,7 +129,11 @@ impl menu::Item for CompletionItem {
                     Style::default()
                 },
             )),
-            menu::Cell::from(kind),
+            if kind.content == "■" {
```

Comment:
When converting to a `Cell` the content is wrapped in two Vecs so avoiding `Spans` above doesn't end up saving any allocations. Instead let's have `kind` be `Spans` and let `.into()` take care of the conversions

Reply:
thanks for letting me know! I did it in https://github.com/helix-editor/helix/pull/12299/commits/52ce26ccecfe359668834c49476a7447a91eabff